core/src/main/java/org/apache/gravitino/listener/api/event/AccessControlEventDispatcher.java

```diff
@@ -346,7 +340,8 @@ public Role createRole(
 
       return roleObject;
     } catch (Exception e) {
-      // TODO: add failure event
+      eventBus.dispatchEvent(
+          new CreateRoleFailureEvent(initiator, metalake, e, role, properties, securableObjects));
```

Comment:
How about creating a roleInfo to represent createRoleRequest with role, properties, securableObjects?